uploadservice/src/main/java/net/gotev/uploadservice/MultipartUploadTask.java

```diff
@@ -143,10 +145,13 @@ private void writeFiles(BodyWriter bodyWriter) throws IOException {
 
     @Override
     protected void onSuccessfulUpload() {
-        for (UploadFile file : params.getFiles()) {
-            addSuccessfullyUploadedFile(file.getPath());
+        // addSuccessfullyUploadedFile removes the file from the list so we have
+        // to use a copy to not get a java.util.ConcurrentModificationException
+        List<UploadFile> files = new ArrayList<>(params.getFiles());
```

Comment:
For that reason  introduced a new method which traverses all the files: https://github.com/gotev/android-upload-service/commit/e055384a56c6e233ec310ed4feac8d40754dda5b